chemprop/models/model.py

```diff
@@ -212,21 +217,44 @@ def predict_step(self, batch: TrainingBatch, batch_idx: int, dataloader_idx: int
         return self(bmg, X_vd, X_d)
 
     def configure_optimizers(self):
-        opt = optim.Adam(self.parameters(), self.init_lr)
-
-        lr_sched = NoamLR(
-            opt,
-            self.warmup_epochs,
-            self.trainer.max_epochs,
-            self.trainer.estimated_stepping_batches // self.trainer.max_epochs,
-            self.init_lr,
-            self.max_lr,
-            self.final_lr,
+        no_decay = ["bias", "bn.weight"]
+        optimizer_grouped_parameters = [
+            {
+                "params": [
+                    p for n, p in self.named_parameters() if not any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": self.weight_decay,
+            },
+            {
+                "params": [
+                    p for n, p in self.named_parameters() if any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": 0.0,
+            },
+        ]
+
+        opt = optim.AdamW(optimizer_grouped_parameters, self.init_lr)
+        steps_per_epoch = self.trainer.num_training_batches
+        warmup_steps = self.warmup_epochs * steps_per_epoch
+        if self.trainer.max_epochs == -1:
+            warnings.warn(
+                "For infinite training, the number of cooldown epochs in learninig rate scheduler is set to 100 times the number of warmup epochs."
+            )
```

Comment:
this seems like a `logging.info` call

Reply:
I use `warnings.warn` because this code is not in the CLI and, therefore, doesn't make use of the logger.

Reply:
what do you mean? All code can make use of the logger. Look at lightning for an example

Reply:
I made this statement because that is how we currently handle it in the codebase, also from [here](https://github.com/chemprop/chemprop/pull/953#issuecomment-2207250068). I am happy to always use the logger if it is the correct way to do it.

Reply:
Good point!